File: angular-course-project/src/app/recipes/recipe-list/recipe.service.ts
import { EventEmitter, Output, Injectable } from '@angular/core';
import { Recipe } from './recipe.model';
import { Ingredient } from '../../shared/ingredient.model';
import { ShoppingListService } from '../../shopping-list/shopping-list.service';

@Injectable()
export class RecipeService{

    constructor(private shoppingListService: ShoppingListService){
    }

    recipes: Recipe[] = [
        new Recipe("Chicken Tikka", 
                    "Indian spicy dish", 
                    "https://glebekitchen.com/wp-content/uploads/2016/12/chickentikkakebab.jpg",
                    [
                        {
                            name: "Chicken",
                            amount: 5
                        },
                        {
                            name: "Tikka Masala",
                            amount: 2  
                        }
                    ]
                ),
        new Recipe("Mutton Biryani", 
                    "Indian Tadka", 
                    "https://image.shutterstock.com/z/stock-photo-spicy-and-delicious-mutton-biryani-790330249.jpg",
                        [
                        {
                            name: "Mutton",
                            amount: 10
                         },
                         {
                             name: "Rice",
                             amount: 5
                         }
                    ]
                )
      ];

    getRecipes(){
        return this.recipes.slice();
    }

    getRecipe(id: number) {
        return this.recipes.splice(1, id);
    }

    addIngredientsToShoppingList(ingredients: Ingredient[]){

        this.shoppingListService.addIngredients(ingredients);
       
    }

}

File: angular-spring-boot-app/server-api/src/main/java/com/learn/angular/services/RecipeService.java
package com.learn.angular.services;

import com.learn.angular.model.Ingredient;
import com.learn.angular.model.Recipe;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class RecipeService {

    public List<Recipe> getRecipes() {

        List<Recipe> recipes = new ArrayList<>();

        List<Ingredient> chickenTikkaIngredients = new ArrayList<>();
        chickenTikkaIngredients.add(new Ingredient("Chicken", 5));
        chickenTikkaIngredients.add(new Ingredient("Tikka Masala", 2));

        Recipe chikenTikkaRecipe = new Recipe(
                "Chicken Tikka",
                "Indian spicy dish",
                "https://glebekitchen.com/wp-content/uploads/2016/12/chickentikkakebab.jpg",
                chickenTikkaIngredients
        );

        List<Ingredient> muttonBiryaniIngredients = new ArrayList<>();
        muttonBiryaniIngredients.add(new Ingredient("Mutton", 10));
        muttonBiryaniIngredients.add(new Ingredient("Rice", 5));

        Recipe muttonBiryaniRecipe = new Recipe(
                "Mutton Biryani",
                "Indian Tadka",
                "https://image.shutterstock.com/z/stock-photo-spicy-and-delicious-mutton-biryani-790330249.jpg",
                muttonBiryaniIngredients
        );

        recipes.add(chikenTikkaRecipe);
        recipes.add(muttonBiryaniRecipe);

        return recipes;
    }
}
